### ISSUE TITLE
Support authentication when checking links

### ISSUE THREAD

**Issue by ryanwinter** [UmbrellaDocs/linkspector#82]
**Is your feature request related to a problem? Please describe.**
I have run this using a private repo (not going public for a few months) and it fails all link checks again the internal repository itself.

**Describe the solution you'd like**
Add the ability to set an authorization token on the link check. This would give the ability to pass a personal access token to the action.

**Describe alternatives you've considered**
Unsure if there is an alternative

**Comment by pwalleni** [comment]
Yes, please. Resources protected by authentication may cause the server to return `404` to linkspector. We want to check if the protected resource exists to ensure the link works when authenticated users click it. We can add `404` to the list of `aliveStatusCodes`, but it's not a great workaround.

**Comment by gaurav-nelson** [comment]
Handling this in #107

**Comment by gaurav-nelson** [comment]
@ryanwinter @pwalleni Can you please verify if this is now fixed with https://github.com/UmbrellaDocs/linkspector?tab=readme-ov-file#http-headers 

Otherwise, I'll need to close the issue. Thank you.

**Comment by gaurav-nelson** [comment]
Done via hhtHeaders https://github.com/UmbrellaDocs/linkspector#configuration
